### DISCUSSION TITLE
[API][Categories] How to generate Category ID

### DISCUSSION THREAD

**Discussion by akama-aka** [stoatchat/discussions#738] [Q&A] [answered]
Hello,
I am currently experimenting with the Revolt API. I noticed that the endpoint to create categories is a PATCH request to the server. The thing is in the PATCH request an id is required but nowhere is it written how to create a normal Revolt Like ID so that the structure of the ID's doesn't look so weird. Does anyone know a scheme for generating the category ID?

**Comment by andrewthederp** [comment, marked as answer]
Revolt id's are usually the date timestamp which the object was created converted into a ULID. However this API design is quite weird and unpractical considering how it defeat the purpose of an ID (unique and never changes)

**Reply by akama-aka** [reply]
Thanks and I can agree with it that the API is a _bit_ weird x3
